components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -241,158 +237,115 @@ public void configure() throws IOException, XMLStreamException, DataEndpointAuth
+                    for (ActivityImpl activity : activities) {
+                        if (activity.getProperty("type").toString().equalsIgnoreCase("usertask")) {
+                            tasks.get(activity.getId())
+                                    .addTaskListener(TaskListener.EVENTNAME_COMPLETE, new TaskCompletionListener());
+                        }
+                        // We are publishing analytics data of service tasks in process termination ATM.
+
+                        // else if(activity.getProperty("type").toString().equalsIgnoreCase("servicetask")){
+                        //       activity.addExecutionListener(PvmEvent.EVENTNAME_END,new
+                        // ServiceTaskCompletionListener());
+                        // }
+                    }
                 }
             }
 
-            // Configure parse handlers, which attaches KPI data publishing listeners to new processes
+            // Configure parse handlers, which attaches data publishing listeners to new processes
             if (log.isDebugEnabled()) {
-                log.debug("Associating parse handlers for processes and tasks, so that KPI data publishing "
-                        + "listeners will be attached to new processes.");
+                log.debug("Associating parse handlers for processes and tasks, so that data publishing listeners "
+                        + "will be attached to new processes.");
             }
-            ProcessEngineConfigurationImpl engineConfig = (ProcessEngineConfigurationImpl) engineService
-                    .getProcessEngine().getProcessEngineConfiguration();
+            ProcessEngineConfigurationImpl engineConfig = (ProcessEngineConfigurationImpl) engineService.
+                    getProcessEngine().getProcessEngineConfiguration();
             if (engineConfig.getPostBpmnParseHandlers() == null) {
                 engineConfig.setPostBpmnParseHandlers(new ArrayList<BpmnParseHandler>());
             }
-            engineConfig.getPostBpmnParseHandlers().add(new ProcessKPIParseHandler());
-            engineConfig.getBpmnDeployer().getBpmnParser().getBpmnParserHandlers()
-                    .addHandler(new ProcessKPIParseHandler());
+            if (genericAnalyticsEnabled) {
+                engineConfig.getPostBpmnParseHandlers().add(new ProcessParseHandler());
+                engineConfig.getPostBpmnParseHandlers().add(new TaskParseHandler());
+            }
+            if (kpiAnalyticsEnabled) {
+                engineConfig.getPostBpmnParseHandlers().add(new ProcessKPIParseHandler());
+                engineConfig.getBpmnDeployer().getBpmnParser().getBpmnParserHandlers()
+                        .addHandler(new ProcessKPIParseHandler());
+            }
+            // engineConfig.getPostBpmnParseHandlers().add(new ServiceTaskParseHandler());
+            engineConfig.getBpmnDeployer().getBpmnParser().getBpmnParserHandlers().
```

Comment:
These handler associations have to be done only if generic analytics is enabled.